ui/app/pages/home/home.component.js

```diff
@@ -106,6 +109,8 @@ export default class Home extends PureComponent {
       history.push(CONFIRM_TRANSACTION_ROUTE);
     } else if (Object.keys(suggestedTokens).length > 0) {
       history.push(CONFIRM_ADD_SUGGESTED_TOKEN_ROUTE);
+    } else if (pendingApprovals.length > 0) {
```

Comment:
We had talked about this going ahead of other confirmations in one of the design calls, but.... on second thought, I think putting it last makes the most sense, as you've done here. This gives the user the opportunity to clear their queue of confirmations first. But more importantly, a dapp can't batch further actions after the network switch anyway because we don't support multi-network connections yet. So this will always be the last action a dapp proposes.

Reply:
Woo hoo. mistakes that work in my favor for the win.